server/game/core/Player.js

```diff
@@ -875,9 +898,20 @@ class Player extends GameObject {
     /**
      * Exhaust the specified number of resources
      */
-    exhaustResources(count) {
+    exhaustResources(count, priorityResources = []) {
+        let priorityResourcesExhausted = 0;
         let readyResources = this.resources.filter((card) => !card.exhausted);
-        for (let i = 0; i < Math.min(count, readyResources.length); i++) {
+        if (priorityResources.length > 0) {
+
+        }
+        priorityResources.forEach((priority) => {
```

Comment:
Had this in the back of my mind because I had a nagging feeling there was something even simpler - realized you could just make an array of all ready resources with the priority ones first, then just do `.slice(0, count).forEach((resource) => resource.exhaust())`

Reply:
@AMMayberry1 How would I get the priority ones into the beginning of the array?

Reply:
Just use filters to generate a list of ready priority resources and the concatenate the list of ready non-priority resources onto it. Or, you could use a filter to generate the list of all ready resources of either type and then sort by priority.

Reply:
Right, but to get the non-priority list, wouldn't I need to filter them or loop anyway?

Reply:
Ah right I keep forgetting it's a list and not a property of the cards themselves. It would still be doable but not worth the effort imo, probably just easiest to go with something like the code example above